File: src/server/Hoard.Core/Application/Portfolios/PositionPerformanceDto.cs
namespace Hoard.Core.Application.Portfolios;

public class PositionPerformanceDto
{
    public decimal CostBasis { get; set; }
    public decimal Units { get; set; }
    
    // Valuation snapshot
    public decimal Value { get; set; }
    public decimal PreviousValue { get; set; }
    public decimal ValueChange { get; set; }
    
    // Gains breakdown
    public decimal UnrealisedGain { get; set; }
    public decimal RealisedGain { get; set; }
    public decimal Income { get; set; }
    
    // Returns
    public decimal? Return1D { get; set; }
    public decimal? Return1W { get; set; }
    public decimal? Return1M { get; set; }
    public decimal? Return3M { get; set; }
    public decimal? Return6M { get; set; }
    public decimal? Return1Y { get; set; }
    public decimal? Return3Y { get; set; }
    public decimal? Return5Y { get; set; }
    public decimal? Return10Y { get; set; }
    public decimal? ReturnYtd { get; set; }
    public decimal? ReturnAllTime { get; set; }
    public decimal? AnnualisedReturn { get; set; }

    public DateTime UpdatedUtc { get; set; }
}

File: src/client/hoard-ui/src/api/dtos/Portfolios/PositionPerformanceDto.ts
export interface PositionPerformanceDto {
  costBasis: number
  units: number

  // Valuation snapshot
  value: number
  previousValue: number
  valueChange: number

  // Gains breakdown
  unrealisedGain: number
  realisedGain: number
  gain: number
  income: number
  profit: number

  // Absolute value changes
  valueChange1Y: number | null

  // Returns (nullable = insufficient history)
  return1D: number | null
  return1W: number | null
  return1M: number | null
  return3M: number | null
  return6M: number | null
  return1Y: number | null
  return3Y: number | null
  return5Y: number | null
  return10Y: number | null
  returnYtd: number | null
  returnAllTime: number | null
  annualisedReturn: number | null

  updatedUtc: string // ISO 8601 DateTime
}
